### PR TITLE
Submission submitted object binding

### LINKED ISSUE #109: PPSubmissionDecorator problem with no relation to submitted PP
Currently when PPsubmission is completed, it is not assigned to the created PP object. It prevents PPSubmissionDecorator.details_path from getting link to the submitted PP object.

Assuming every submission will submit one "main" (or "parent") object, perhaps we may solve it by having a single "generic" relation id to a submitted object? Still, it is not clear at the moment if this is a good solution, so for the time being we need to see what happens in further submission processes.

### PR DESCRIPTION
**Pull request by Nuanda**
Fixes #109 

Fix is generic, should work for future submission types as well.

### CODE REVIEW

File: app/models/submission.rb [hunk 1]
```diff
@@ -58,6 +63,21 @@ def name
     [I18n.l(created_at, format: :short), content.step01.try(:name)].compact.join(' ')
   end
 
+  def associated_model
+    case submission_type
+      when 'population'
```
Comment (kammerer) [inline app/models/submission.rb]: Strange indentation ;)

File: spec/factories/submission.rb [hunk 2]
```diff
@@ -2,5 +2,12 @@
   factory :submission do
     submission_type :population
     user
+
+    trait :finalized do
```
Comment (kammerer) [inline spec/factories/submission.rb]: I introduced such trait too in #178 but it is more havy-weight  as it assigns `submission.content` and performs finalization. I think it is worth to keep it but maybe under different name.

**Comment by kammerer** [comment]
:+1:

File: app/models/submission.rb (hunk 1 above)
Comment (Nuanda) [inline app/models/submission.rb]: Fixed.

File: spec/factories/submission.rb (hunk 2 above)
Comment (Nuanda) [inline spec/factories/submission.rb]: Let's do it like that: I merge my PR and when merging yours, you will add what is important (mainly: setting submitted_object_id) from my trait to your (better) trait, and after all specs are ok, you'll remove my trait.

**Pull request merged**